pubtools/_pulp/hooks.py

```diff
@@ -14,4 +14,15 @@ def task_pulp_flush():
     Pulp content may have recently changed."""
 
 
+@hookspec
+def pulp_item_finished(
+    item_metadata=None, push_item=None
+):  # pylint: disable=unused-argument
+
+    """Invoked when PulpPushItem is pushed to a pulp target
+
+    This hook should be invoked per every pub push item after end of push
+    process."""
```

Comment:
- The doc should be fleshed out a bit I think, also rewritten because PulpPushItem is not a thing which makes sense anywhere outside of this code.
- Concrete types should probably be documented as well.
- Later in the change I'm suggesting to yield actual unit and not a dict, so the name of that argument should change.
- As suggested I think this should have a list of units, not a single unit, to reserve the possibility of an item generating multiple units.  modulemd and comps.xml are both current examples of push items which end up creating multiple units (though not tracked by pubtools-pulp today).
- And finally we only invoke this hook for push tasks, and not other kinds of tasks which can also process "items" (like publish or fix CVEs), so the hook name should probably reflect that.

So overall I suggest following changes:

```suggestion
def pulp_item_push_finished(
    pulp_units=None, push_item=None
):  # pylint: disable=unused-argument
    """Invoked during push tasks after each item has been processed fully.
    
    By the time this hook is invoked, the referenced item and unit is expected
    to be fully uploaded into Pulp and published onto the CDN.

    Args:
        pulp_units (list[:class:`~pubtools.pulplib.Unit`])
            A list of zero or more Pulp unit(s) created/updated for this item.

            Note that this information may not be available for every content type,
            and may only contain a subset of the Pulp fields.

        push_item (:class:`~pushsource.PushItem`)
            The item which has been pushed.
    """
```